### PR TITLE
fix(server): only workspaces with no projects are removable

### PR DESCRIPTION
**Pull request by nourbalaha**
In this PR:
1- I added the functionality to check and make sure that the workspace has no projects before removing it. so, only workspaces with no projects are removable.
2- I increased the code coverage for workspace and project.

### CODE REVIEW

File: server/internal/usecase/interactor/workspace_test.go [hunk 1]
```diff
@@ -57,6 +64,7 @@ func TestWorkspace_Fetch(t *testing.T) {
 			operator *usecase.Operator
 		}
 		want    []*user.Workspace
+		mockErr bool
```
Comment (rot1024) [inline server/internal/usecase/interactor/workspace_test.go:67]:
```suggestion
		mockWorkspaceErr bool
```

File: server/internal/usecase/interactor/workspace_test.go [hunk 2]
```diff
@@ -111,13 +119,25 @@ func TestWorkspace_Fetch(t *testing.T) {
...
 
+			r := &Workspace{}
+			if tc.mockErr {
+				SetWorkspaceError(r, tc.wantErr)
+				assert.Equal(t, tc.wantErr, r.err)
+				return
+			}
+
 			ctx := context.Background()
 			db := memory.New()
```
Comment (rot1024) [inline server/internal/usecase/interactor/workspace_test.go:142]:
```suggestion
			db := memory.New()
			if tc.mockWorkspaceErr {
				memory.SetWorkspaceError(db.Workspace, tc.wantErr)
			}
```

File: server/internal/usecase/interactor/workspace.go [hunk 3]
```diff
@@ -40,7 +41,7 @@ func (i *Workspace) FindByUser(ctx context.Context, id id.UserID, operator *usec
 }
 
 func (i *Workspace) Create(ctx context.Context, name string, firstUser id.UserID, operator *usecase.Operator) (_ *user.Workspace, err error) {
-	return Run1(ctx, operator, i.repos, Usecase().Transaction(), func() (*user.Workspace, error) {
+	return Run1(ctx, operator, i.repos, Usecase().Transaction(), func() (*user.Workspace, error) {
```
Comment (yk-eukarya) [inline server/internal/usecase/interactor/workspace.go:44]:
```suggestion
	return Run1(ctx, operator, i.repos, Usecase().Transaction(), func() (*user.Workspace, error) {
```

Comment (yk-eukarya) [reply server/internal/usecase/interactor/workspace.go:44]: don't forget to run the code linter or configure your IDE to use it

File: server/internal/usecase/interactor/workspace.go [hunk 4]
```diff
@@ -201,6 +201,14 @@ func (i *Workspace) Remove(ctx context.Context, id id.WorkspaceID, operator *use
 			return interfaces.ErrOperationDenied
 		}
 
+		projectCount, err := i.repos.Project.CountByWorkspace(ctx, id)
+		if err != nil {
+			return err
+		}
+		if projectCount > 0 {
+			return user.ErrWorkspaceWithProjects
```
Comment (rot1024) [inline server/internal/usecase/interactor/workspace.go:209]:
```suggestion
			return interfaces.ErrWorkspaceWithProjects
```

This error is not a user-related error. Define it at `internal/usecase/interfaces/workspace.go`

**Review by rot1024 — APPROVED** [review]

**Pull request merged**